src/pages/SearchPage/SearchPage.test.js

```diff
@@ -0,0 +1,203 @@
+  const url = `https://api.pushshift.io/reddit/search/submission/?${DEFAULT_PATH}=${DEFAULT_SUBREDDIT}&sort=desc&sort_type=score&after=${oneYearAgo}&size=500`;
+  if (mockState === 'pass') {
+    axiosMock.get.mockResolvedValueOnce({ data });
+  } else if (mockState === 'fail') {
+    axiosMock.get.mockRejectedValue({ data });
+  }
+  const { getByTestId, getAllByTestId, queryByTestId } = renderWithRouter(
+    <SearchPage />, DEFAULT_PATH, DEFAULT_SUBREDDIT,
+  );
+  return {
+    getByTestId,
+    getAllByTestId,
+    queryByTestId,
+    url,
+  };
+}
+
+function getHeatmapValues(dataArray) {
+  const parsedData = parseRedditData(dataArray);
+  return parsedData.map((day) => day.map((hour) => hour.length)).flat();
+}
+
+describe('Search controls', () => {
+  test('Default input is used when search page is loaded', async () => {
+    const { getByTestId } = setup();
+    const input = getByTestId('search-input');
+    expect(input).toBeInTheDocument();
+    expect(input.value).toEqual(DEFAULT_SUBREDDIT);
+    await waitForElementToBeRemoved(getByTestId('loading-spinner'));
+    expect(axiosMock.get).toHaveBeenCalledTimes(1);
+  });
+  test('Can change the subreddit being searched', async () => {
+    const NEW_SUBREDDIT = 'learnprogramming';
+    const { getByTestId } = setup('pass', dummyPosts);
+    const button = getByTestId('search-button');
+    const input = getByTestId('search-input');
+    expect(input).toBeInTheDocument();
+    expect(input.value).toEqual(DEFAULT_SUBREDDIT);
+    expect(axiosMock.get).toBeCalledWith(expect.stringMatching(DEFAULT_SUBREDDIT));
+    await waitForElementToBeRemoved(getByTestId('loading-spinner'));
+    fireEvent.change(input, { target: { value: NEW_SUBREDDIT } });
+    expect(input.value).toEqual(NEW_SUBREDDIT);
+    fireEvent.click(button);
+    await waitForElementToBeRemoved(getByTestId('loading-spinner'));
+    expect(axiosMock.get).toHaveBeenCalledTimes(2);
+    expect(axiosMock.get).toBeCalledWith(expect.stringMatching(NEW_SUBREDDIT));
+  });
+  test('Search button is disabled when loading takes place', async () => {
+    const { getByTestId } = setup();
+    const button = getByTestId('search-button');
+    const input = getByTestId('search-input');
+    expect(input).toBeInTheDocument();
+    expect(input.value).toEqual(DEFAULT_SUBREDDIT);
+    expect(button).toBeDisabled();
+    await waitForElementToBeRemoved(getByTestId('loading-spinner'));
+    expect(axiosMock.get).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('Error and heatmap display states', () => {
+  test('Will show an error message when there are no results', async () => {
+    const { getByTestId } = setup();
+    expect(getByTestId('loading-spinner')).toBeInTheDocument();
+    await waitForElementToBeRemoved(getByTestId('loading-spinner'));
+    const error = getByTestId('error');
+    expect(error).toBeInTheDocument();
+    expect(error.textContent).toBe('0 results returned.');
```

Comment:
Instead of these lines you could write

```
await findByText('0 results returned.');
```

With this assertion you test the spinner over and over again. It would be enough to do that once and only use `waitForElementToBeRemoved` if required

```
expect(getByTestId('loading-spinner')).toBeInTheDocument();
await waitForElementToBeRemoved(getByTestId('loading-spinner'));
```